- basic configuration for adding videos to firestore

smlltt opened the pull request:
(https://firebase.google.com/docs/firestore/quickstart)
- so far I only created a collection (videos) as in the instructions above
- maybe it makes sense to explicity define the document identifier instead of having a randomly generated code as explained here (the doc identifier could be the video id). let me know: https://firebase.google.com/docs/firestore/manage-data/add-data

- before adding a video, I check if the video exists already and show a warning. this can and probably should be changed: if the video already exists (added by someone else), we can show a massage such as "video added to your collection" (it will not be added to the backend as a new video, because it already exists), but the user property will be modified, so that the video is now linked both to the original uploader and the new one. For example I could change the videoObject like so: instead of userId we would have "originalUploaderId" and add a new category: likedBy: [originalUploaderId, newGuyThatTriedToUploadTheSameVideo, someGuyThatLikedTheVideo, etc]

TO BE DONE:
- redux's store will take it's video list values from firestore

ZachariaszStrus committed:
- basic configuration for adding videos to firestore
(https://firebase.google.com/docs/firestore/quickstart)
- so far I only created a collection (videos) as in the instructions above
- maybe it makes sense to explicity define the document identifier instead of having a randomly generated code as explained here (the doc identifier could be the video id). let me know: https://firebase.google.com/docs/firestore/manage-data/add-data

- before adding a video, I check if the video exists already and show a warning. this can and probably should be changed: if the video already exists (added by someone else), we can show a massage such as "video added to your collection" (it will not be added to the backend as a new video, because it already exists), but the user property will be modified, so that the video is now linked both to the original uploader and the new one. For example I could change the videoObject like so: instead of userId we would have "originalUploaderId" and add a new category: likedBy: [originalUploaderId, newGuyThatTriedToUploadTheSameVideo, someGuyThatLikedTheVideo, etc]

TO BE DONE:
- redux's store will take it's video list values from firestore

ZachariaszStrus commented on src/screens/AddVideo/AddVideo.component.tsx:
```diff
@@ -6,15 +6,25 @@ export type Level = 'beginner' | 'intermediate' | 'advanced' | '';
 interface OwnProps {
   addVideo(video: AddVideoFormUrl): void;
+  videoIsDuplicate: boolean;
+  setVideoIsDuplicateToFalse: () => void;
```
i'd call this props 'onCancel' (also in AddVideoFormik) - it's more generic

ZachariaszStrus commented:
the form should be reset after successful submission, (also we could use some visual info for user - can be done later)

ZachariaszStrus commented on src/screens/AddVideo/index.tsx:
```diff
@@ -1,16 +1,30 @@
+    setVideoIsDuplicate(false);
+  };
+
   const youTubeApiCall = (
```
i'd move this function to new youTubeService (like firebase)

ZachariaszStrus replied:
i'd create dedicated function in fireApi for adding video, so it can be typed properly

ZachariaszStrus committed:
- when you add a video, you get a confirmation message. the form is cleared;
- when you start typing in order to add a new video, the confirmation message is cleared. Same when video already exists (warning message disappears)
- firestore logic moved to services, firebase

TO BE DONE:
- redux's store will take it's video list values from firestore

ZachariaszStrus committed:
- setVideoIsDuplicateToTrue changed to onCancel
- setVideoIsDuplicateToFalse changed to onConfirm

TO BE DONE:
- redux's store will take it's video list values from firestore

ZachariaszStrus committed:
- setVideoIsDuplicateToFalse changed to onCancel

TO BE DONE:
- redux's store will take it's video list values from firestore
